ArkhamOverlay/Pages/Overlay/OverlayController.cs

```diff
@@ -337,15 +345,43 @@ public class OverlayController : Controller<OverlayView, OverlayViewModel> {
             }
         }
 
+        private void AutoSnapshot() {
+            //every half second we write a snapshot to a temp file and then try to overwrite the configured file with it
+            //this prevents OBS from getting in a weird state where it tries to read a file that is not there (because we are writing it)
+            var tempFileName = Path.GetTempFileName();
+            WriteSnapshotToFile(tempFileName);
+
+            //only overwrite the overlay file if there has been a change
+            if (FileCompare.CompareFiles(tempFileName, _appData.Configuration.AutoSnapshotFilePath)) {
```

Comment:
If we already have the new file (as a temp), does not overwriting the old one matter? I feel like we're just wasting time comparing the two files

Reply:
It matters in the case where OBS tries to open a file we are writing to (or copying over). It will then hide the overlay for a few seconds. This happens very rarely, but there are less odds of it happning the less we write. Thus, we should only write when there are actually changes.